File: task-03/prime.js



function is_prime(num) {
    var count = 0;
    for (let i = 1; i <= num; i++) {
        if (num % i === 0) {
            count+=1;
        }
    }
    if(count === 2)
        return true;
    else
        return false;
}


function main() {
    const n = parseInt(prompt("Enter a number:"));
    for (let i = 2; i <= n; i++) {
        if (is_prime(i)) {
            console.log(i);
        }
    }
    
}

main();

File: task-03/prime.hs



isPrime :: Int -> Bool
isPrime x = count == 2
  where
    count = length [i | i <- [1..x], x `mod` i == 0]

main :: IO ()
main = do
  putStrLn "Enter the limit: "
  num <- readLn :: IO Int
  let primes = [i | i <- [2..num], isPrime i]
  mapM_ print primes
